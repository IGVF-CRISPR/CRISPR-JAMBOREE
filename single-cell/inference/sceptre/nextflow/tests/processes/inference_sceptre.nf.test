nextflow_process {

    name "Test Process inference_sceptre"
    script "processes/inference_sceptre.nf"
    process "inference_sceptre"

    test("inference_sceptre on Gasperini data with default parameters") {

        when {
            params {
                side = "both"
                grna_integration_strategy = "union"
                resampling_approximation = "skew_normal"
                control_group = "default"
                resampling_mechanism = "default"
                formula_object = "default"
            }
            process {
                """
                input[0] = file("${projectDir}/data/gasperini_inference_input.h5mu")
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process).match() }
            )
        }
    }

    test("inference_sceptre on Gasperini data with alternative parameters") {

        when {
            params {
                side = "left" // different side
                grna_integration_strategy = "union"
                resampling_approximation = "skew_normal"
                control_group = "default"
                resampling_mechanism = "default"
                formula_object = "~ prep_batch + log(response_n_nonzero) + log(response_n_umis)" // different formula object
            }
            process {
                """
                input[0] = file("${projectDir}/data/gasperini_inference_input.h5mu")
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process).match() }
            )
        }
    }

    test("inference_sceptre on Papalexi data with default parameters") {

        when {
            params {
                side = "both"
                grna_integration_strategy = "union"
                resampling_approximation = "skew_normal"
                control_group = "default"
                resampling_mechanism = "default"
                formula_object = "default"
            }
            process {
                """
                input[0] = file("${projectDir}/data/papalexi_inference_input.h5mu")
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process).match() }
            )
        }
    }

    test("inference_sceptre on Papalexi data with alternative parameters") {

        when {
            params {
                side = "both" 
                grna_integration_strategy = "union"
                resampling_approximation = "skew_normal"
                control_group = "complement" // different control group
                resampling_mechanism = "default"
                formula_object = "~ replicate + log(response_n_nonzero) + log(response_n_umis)" // different formula object
            }
            process {
                """
                input[0] = file("${projectDir}/data/papalexi_inference_input.h5mu")
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process).match() }
            )
        }
    }
}