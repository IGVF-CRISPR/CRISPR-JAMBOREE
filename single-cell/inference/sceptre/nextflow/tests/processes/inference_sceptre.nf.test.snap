{
    "inference_sceptre on Gasperini data with alternative parameters": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "out": {
                    "0": [
                        "mudata_out.h5mu:md5,14a5f90d60b1eecd9bd011d01998a573"
                    ]
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 1,
                    "tasksSucceeded": 1
                },
                "name": "inference_sceptre",
                "success": true
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T20:20:33.143148"
    },
    "inference_sceptre on Gasperini data with default parameters": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "out": {
                    "0": [
                        "mudata_out.h5mu:md5,3809f36dfacf4c834ccf22249254f2bf"
                    ]
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 1,
                    "tasksSucceeded": 1
                },
                "name": "inference_sceptre",
                "success": true
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T20:20:00.382146"
    },
    "inference_sceptre on Papalexi data with alternative parameters": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "out": {
                    "0": [
                        "mudata_out.h5mu:md5,bc8207f3d3a1bbd194b1505c15c29d31"
                    ]
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 1,
                    "tasksSucceeded": 1
                },
                "name": "inference_sceptre",
                "success": true
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T20:21:20.284628"
    },
    "inference_sceptre on Papalexi data with default parameters": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "out": {
                    "0": [
                        "mudata_out.h5mu:md5,9c81a4b859c2fddd61ae17331b470846"
                    ]
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 1,
                    "tasksSucceeded": 1
                },
                "name": "inference_sceptre",
                "success": true
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T20:20:54.02798"
    }
}